apps/server/lib/lexical/server/project/diagnostics.ex

```diff
@@ -133,20 +135,23 @@ defmodule Lexical.Server.Project.Diagnostics do
     end
 
     defp position_to_range(%SourceFile{} = source_file, {line_number, column}) do
-      line_number = Math.clamp(line_number - 1, 0, SourceFile.size(source_file) - 1)
-      column = max(column - 1, 0)
-
-      with {:ok, line_text} <- SourceFile.fetch_text_at(source_file, line_number),
-           {:ok, character} <- CodeUnit.to_utf16(line_text, column) do
-        Range.new(
-          start: Position.new(line: line_number, character: character),
-          end: Position.new(line: line_number + 1, character: 0)
+      line_number = Math.clamp(line_number, 1, SourceFile.size(source_file) + 1)
```

Comment:
why changed `SourceFile.size(source_file) - 1` to `SourceFile.size(source_file) + 1)`?

We need to ensure that the line_number does not exceed the actual maximum number of lines, so we need to subtract 1.